pkg/awsutils/awsutils.go

```diff
@@ -387,15 +392,58 @@ func (cache *EC2InstanceMetadataCache) refreshSGIDs(mac string) error {
 	newSGs := StringSet{}
 	newSGs.Set(sgIDs)
 	addedSGs := newSGs.Difference(&cache.securityGroups)
+	addedSGsCount := 0
 	deletedSGs := cache.securityGroups.Difference(&newSGs)
+	deletedSGsCount := 0
 
 	for _, sg := range addedSGs.SortedList() {
 		log.Infof("Found %s, added to ipamd cache", sg)
+		addedSGsCount++
 	}
 	for _, sg := range deletedSGs.SortedList() {
 		log.Infof("Removed %s from ipamd cache", sg)
+		deletedSGsCount++
 	}
 	cache.securityGroups.Set(sgIDs)
+
+	if addedSGsCount != 0 || deletedSGsCount != 0 {
```

Comment:
If "Custom Networking" is enabled, I believe we shouldn't do any of this since then the security groups are explicitly configured. We need to skip that here. Probably even worth having a unit test for that. 🙂

Reply:
Sure Claes will add the check :)